safekeeper/src/handler.rs

```diff
@@ -149,19 +162,28 @@ impl postgres_backend::Handler for SafekeeperPostgresHandler {
         self.check_permission(Some(tenant_id))?;
         self.ttid = TenantTimelineId::new(tenant_id, timeline_id);
 
-        match cmd {
+        let res = match cmd {
             SafekeeperPostgresCommand::StartWalPush => ReceiveWalConn::new(pgb).run(self),
             SafekeeperPostgresCommand::StartReplication { start_lsn } => {
                 ReplicationConn::new(pgb).run(self, pgb, start_lsn)
             }
             SafekeeperPostgresCommand::IdentifySystem => self.handle_identify_system(pgb),
             SafekeeperPostgresCommand::JSONCtrl { ref cmd } => handle_json_ctrl(self, pgb, cmd),
-        }
-        .context(format!(
-            "Failed to process query for timeline {timeline_id}"
-        ))?;
+        };
 
-        Ok(())
+        match res {
+            Ok(()) => Ok(()),
+            Err(MaybeIoError::Io(io_error)) => {
+                info!(
+                    "Timeline {}/{} query failed with IO error: {}",
+                    tenant_id, timeline_id, io_error
+                );
+                Err(MaybeIoError::Io(io_error))
+            }
+            Err(MaybeIoError::Anyhow(e)) => Err(MaybeIoError::Anyhow(e.context(format!(
+                "Failed to process query for timeline {timeline_id}"
```

Comment:
Recently we decided to log tenant_id+timeline_id everywhere, so let's patch it right here:

```suggestion
                "Failed to process query for timeline {self.ttid}"
```